main.tf

```diff
@@ -1,39 +1,33 @@
 module "alb" {
-  count = var.app_type == "web" ? 1 : 0
+  count = var.app_type == "web" || var.app_type == "tcp-app" ? 1 : 0
 
   source  = "registry.terraform.io/terraform-aws-modules/alb/aws"
   version = "~> 5.0"
 
   name               = var.public ? local.name : "${local.name}-private"
-  load_balancer_type = "application"
+  load_balancer_type = var.app_type == "web" ? "application" : "network" 
   internal           = var.public ? false : true
   vpc_id             = var.vpc_id
   security_groups    = var.alb_security_groups
   subnets            = var.public ? var.public_subnets : var.private_subnets
   idle_timeout       = var.alb_idle_timeout
 
-  http_tcp_listeners = [
-    {
-      port               = var.http_port
-      protocol           = "HTTP"
-      target_group_index = 0
-    },
-  ]
+  http_tcp_listeners = local.http_tcp_listeners
 
   https_listeners = var.https_enabled ? concat(
-    [
-      {
-        port               = 443
-        protocol           = "HTTPS"
-        certificate_arn    = var.tls_cert_arn
-        target_group_index = 0
-      }
-    ]) : []
+  [
+    {
+      port               = 443
+      protocol           = var.app_type == "web" ? "HTTPS" : "TLS"
+      certificate_arn    = var.tls_cert_arn
+      target_group_index = 0
+    }
+  ]) : []
```

Comment:
@igorkotof How would it work with multiple ports in a target group? (like 4444,4445 etc)?

Reply:
@AutomationD pls take a look into: `http_tcp_listeners = local.http_tcp_listeners`